`catching polymorphic type` warnings with gcc 8.1

Of course those can be ignored with `-Dwerror=false`, but that isn't feasible.

```
The Meson build system
Version: 0.43.0
Source dir: libzim-3.3.0
Build dir: libzim-3.3.0/build
Build type: native build
Project name: libzim
Native C compiler: cc (gcc 8.1.0)
Native C++ compiler: c++ (gcc 8.1.0)
Build machine cpu family: x86_64
Build machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (0.29.2)
Native dependency zlib found: YES 1.2.11
Native dependency liblzma found: YES 5.2.4
Dependency xapian-core found: NO
Dependency threads found: YES
Native dependency icu-i18n found: YES 61.1
Configuring config.h using configuration
Dependency GTest found: YES (prebuilt)
Program cython3 found: NO
Program nonexistingprogram found: NO
Native dependency python3 found: YES 3.6
Build targets in project: 7
Found ninja-1.8.2 at /usr/bin/ninja
```

```
[15/39] Compiling C++ object 'src/zim@sha/fileimpl.cpp.o'.
FAILED: src/zim@sha/fileimpl.cpp.o 
c++  -Isrc/zim@sha -Isrc -I../src -Iinclude -I../include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++11 -O0 -g -fPIC -pthread -MMD -MQ 'src/zim@sha/fileimpl.cpp.o' -MF 'src/zim@sha/fileimpl.cpp.o.d' -o 'src/zim@sha/fileimpl.cpp.o' -c ../src/fileimpl.cpp
../src/fileimpl.cpp: In constructor ‘zim::FileImpl::FileImpl(const string&)’:
../src/fileimpl.cpp:74:33: error: catching polymorphic type ‘class zim::ZimFileFormatError’ by value [-Werror=catch-value=]
     } catch (ZimFileFormatError e) {
                                 ^
../src/fileimpl.cpp: In member function ‘std::shared_ptr<const zim::Dirent> zim::FileImpl::getDirent(zim::article_index_t)’:
../src/fileimpl.cpp:301:18: error: catching polymorphic type ‘class zim::InvalidSize’ by value [-Werror=catch-value=]
         } catch (InvalidSize) {
                  ^~~~~~~~~~~
../src/fileimpl.cpp: In member function ‘std::__cxx11::string zim::FileImpl::getChecksum()’:
../src/fileimpl.cpp:496:14: error: catching polymorphic type ‘class zim::BufferError’ by value [-Werror=catch-value=]
     } catch (BufferError)
              ^~~~~~~~~~~
cc1plus: all warnings being treated as errors
[18/39] Compiling C++ object 'src/zim@sha/search.cpp.o'.
ninja: build stopped: subcommand failed.
```

--- a/src/fileimpl.cpp
+++ b/src/fileimpl.cpp
@@ -71,7 +71,7 @@ namespace zim
     }
     try {
       header.read(zimReader->get_buffer(offset_t(0), zsize_t(Fileheader::size)));
-    } catch (ZimFileFormatError e) {
+    } catch (ZimFileFormatError& e) {
       throw e;
     } catch (...) {
       throw ZimFileFormatError("error reading zim-file header.");
@@ -298,7 +298,7 @@ namespace zim
         auto direntBuffer = std::unique_ptr<Buffer>(new MemoryBuffer<false>(bufferDirentZone.data(), bufferSize));
         try {
           dirent = std::make_shared<const Dirent>(std::move(direntBuffer));
-        } catch (InvalidSize) {
+        } catch (InvalidSize&) {
           // buffer size is not enougth, try again :
           bufferSize += 256;
           continue;
@@ -493,7 +493,7 @@ namespace zim
     std::shared_ptr<const Buffer> chksum;
     try {
       chksum = zimReader->get_buffer(offset_t(header.getChecksumPos()), zsize_t(16));
-    } catch (BufferError)
+    } catch (BufferError&)
     {
       log_warn("error reading checksum");
       return std::string();

